libs/features/personalization/promo-utils.js

```diff
@@ -1,36 +1,70 @@
-import { getMetadata } from '../../utils/utils.js';
+import { getMetadata, getConfig } from '../../utils/utils.js';
 
 const GMTStringToLocalDate = (gmtString) => new Date(`${gmtString}+00:00`);
 
+const APAC = ['au', 'cn', 'hk_en', 'hk_zh', 'id_en', 'id_id', 'in', 'in_hi', 'kr', 'my_en', 'my_ms', 'nz', 'ph_en', 'ph_fil', 'sg', 'th_en', 'th_th', 'tw', 'vn_en', 'vn_vi'];
+const EMEA = ['ae_en', 'ae_ar', 'africa', 'at', 'be_en', 'be_fr', 'be_nl', 'bg', 'ch_de', 'ch_fr', 'ch_it', 'cis_en', 'cis_ru', 'cz', 'de', 'dk', 'ee', 'eg_ar', 'eg_en', 'es', 'fi', 'fr', 'gr_el', 'gr_en', 'hu', 'ie', 'il_en', 'il_he', 'iq', 'is', 'it', 'kw_ar', 'kw_en', 'lt', 'lu_de', 'lu_en', 'lu_fr', 'lv', 'mena_ar', 'mena_en', 'ng', 'nl', 'no', 'pl', 'pt', 'qa_ar', 'qa_en', 'ro', 'ru', 'sa_en', 'sa_ar', 'se', 'si', 'sk', 'tr', 'ua', 'uk', 'za'];
+const AMERICAS = ['us', 'ar', 'br', 'ca', 'ca_fr', 'cl', 'co', 'cr', 'ec', 'gt', 'la', 'mx', 'pe', 'pr'];
+const JP = ['jp'];
+const REGIONS = { APAC, EMEA, AMERICAS, JP };
+
+const getLocaleCode = () => {
+  const getLocale = () => getConfig()?.locale?.prefix?.substring(1) || 'us';
+  return (getLocale()?.split('-')?.pop() || 'us').toLowerCase();
+};
+
+const localeCode = getLocaleCode();
+const regionCode = Object.keys(REGIONS)
+  .find((r) => REGIONS[r]?.includes(localeCode))?.toLowerCase() || null;
+
 export const isDisabled = (event, searchParams) => {
   if (!event) return false;
+  if (event.locales && !event.locales.includes(localeCode)) return true;
   const currentDate = searchParams?.get('instant') ? new Date(searchParams.get('instant')) : new Date();
   if ((!event.start && event.end) || (!event.end && event.start)) return true;
   return Boolean(event.start && event.end
     && (currentDate < event.start || currentDate > event.end));
 };
 
-export default function getPromoManifests(manifestNames, searchParams) {
+const getRegionalPromoManifests = (manifestNames, region, searchParams) => {
   const attachedManifests = manifestNames
     ? manifestNames.split(',')?.map((manifest) => manifest?.trim())
     : [];
-  const schedule = getMetadata('schedule');
+
+  const schedule = getMetadata(region ? `${region}_schedule` : 'schedule');
   if (!schedule) {
     return [];
   }
   return schedule.split(',')
     .map((manifest) => {
-      const [name, start, end, manifestPath] = manifest.trim().split('|').map((s) => s.trim());
+      const [name, start, end, manifestPath, locales] = manifest.trim().split('|').map((s) => s.trim());
       if (attachedManifests.includes(name)) {
         const event = {
           name,
           start: GMTStringToLocalDate(start),
           end: GMTStringToLocalDate(end),
         };
+        if (locales) {
+          event.locales = locales?.split(';').map((locale) => locale.trim());
```

Comment:
i think we should check here if current page is not in target locales.
and if its not, it should not have an event.
currently you create a disabled event, which means it would show on all unaffected GEOs in the MEP Preview.
instead we could just have:

```
const localesList = locales?.split(';').map((locale) => locale.trim().toLowerCase());
if (!localesList.includes(localeCode)) return true;
```
(I added toLowerCase, it doesn't hurt)
I created https://mwpw-147206--milo--npeltier.hlx.page/de/npeltier/geopromos/promo to demonstrate what i mean, but it doesn't show preview because of another problem
[jp promotions](https://adobe.sharepoint.com/:x:/r/sites/adobecom/_layouts/15/Doc.aspx?sourcedoc=%7B0AF3704A-08DE-455D-8CAF-F38A205362A8%7D&file=promotions-jp.xlsx&action=default&mobileredirect=true) file have wrong column names  (emea) and rewrite metadata for emea

Reply:
i think i do agree with you. Fix the file you created and this is weird to see on de page
<img width="324" alt="image" src="https://github.com/adobecom/milo/assets/1032754/c2a352f2-0400-4420-9a48-d0a7186cc553">
 i fixed it, please have another look @3ch023